src/controller/filter.ts

```diff
@@ -0,0 +1,68 @@
+import { FilterModel, FilterModelProperty } from "../models/filter/filter-model";
+import { FilterConfig } from "../models/filter/filter-config-item";
+import {FilterView} from "../views/interfaces";
+import {EventManager} from "../event-manager/event-manager";
+
+export class Filter<T> extends EventManager {
+    model: FilterModel<T>;
+    view: FilterView;
+
+    constructor(model: FilterModel<T>, view: FilterView) {
+        super();
+        this.model = model;
+        this.view = view;
+
+        document.querySelector(this.view.selector).addEventListener('submit', this.submitEventHandler);
+        document.querySelector(this.view.selector).addEventListener('click', this.clickEventHandler);
+        document.querySelector(this.view.selector).addEventListener('keydown', this.keypressEventHandler);
+        document.querySelector(this.view.selector).addEventListener('focusout', this.focusoutEventHandler);
+    }
+
+    public initNewData (data: T[], filterModel: FilterConfig): void {
+        this.model.initNewData(data, filterModel);
+        this.view.render(this.model.getFilterModel());
+    }
+
+    private submitEventHandler = (e: Event):void => {
+        e.preventDefault();
+        e.stopPropagation();
+        let elem = e.target as HTMLFormElement;
+        for (let i = 0; i < elem.length - 1; i++) {
+            let input = elem[i] as HTMLInputElement; 
+            this.model.setFilterModelProperty(input.dataset['property'], input.dataset['use'] as FilterModelProperty ,+input.value);
```

Comment:
There is no need to use `dataset` because browser supports simple custom attributes. For example: `<div property="displays"></div>`